### ISSUE TITLE
Indent rule with object and array elements

### ISSUE THREAD

**Issue by targos** [eslint/eslint#3237]
Version 1.0.0

Rule:
```yml
indent: [2, 2, {"SwitchCase": 1, "VariableDeclarator": 2}]
```

Code:
```js
var data = {
    x: 0,
    y: 0
};

var arr = [
    0,
    1
];
```

Error:
```
test3.js
  4:2  error  Expected indentation of 2 characters but found 0  indent
  9:2  error  Expected indentation of 2 characters but found 0  indent
```

I expect the rule to warn about the indentation of array and object elements (should be 2 instead of 4), but it is flagging the last line of the literal instead.
Note: when indentation of the elements is correct, there is no problem.

Cross-reference: https://github.com/nodejs/io.js/pull/2286

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by gyandeeps** [comment]
what is expected when we have multiple variable declarations.
rule config: `[2, 2, {"VariableDeclarator": 2, "SwitchCase": 1}]` 

A:
```js
var data = {
  x: 0,
  y: 0
},
    b = 2;
```

B:
```js
var data = {
      x: 0,
      y: 0
    },
    b = 2;
```

Based on above examples, which is correct `A` or `B`?

**Comment by IanVS** [comment]
My vote is for `B`.

**Comment by targos** [comment]
I'm also for `B`.

**Comment by nzakas** [comment]
B for sure
